pkg/apis/componentconfig/types.go

```diff
@@ -164,171 +164,211 @@ type GroupResource struct {
+	// ResourceQuotaControllerConfiguration holds configuration for
+	// ResourceQuotaController related features.
+	ResourceQuotaController ResourceQuotaControllerConfiguration
+	// SAControllerConfiguration holds configuration for ServiceAccountController
+	// related features.
+	SAController SAControllerConfiguration
 	// Controllers is the list of controllers to enable or disable
 	// '*' means "all enabled by default controllers"
 	// 'foo' means "enable 'foo'"
 	// '-foo' means "disable 'foo'"
 	// first item for a particular name wins
 	Controllers []string
-
-	// port is the port that the controller-manager's http service runs on.
-	Port int32
-	// address is the IP address to serve on (set to 0.0.0.0 for all interfaces).
-	Address string
-	// useServiceAccountCredentials indicates whether controllers should be run with
-	// individual service account credentials.
-	UseServiceAccountCredentials bool
-	// cloudProvider is the provider for cloud services.
-	CloudProvider string
-	// cloudConfigFile is the path to the cloud provider configuration file.
-	CloudConfigFile string
 	// externalCloudVolumePlugin specifies the plugin to use when cloudProvider is "external".
 	// It is currently used by the in repo cloud providers to handle node and volume control in the KCM.
 	ExternalCloudVolumePlugin string
-	// run with untagged cloud instances
-	AllowUntaggedCloud bool
-	// concurrentEndpointSyncs is the number of endpoint syncing operations
-	// that will be done concurrently. Larger number = faster endpoint updating,
-	// but more CPU (and network) load.
-	ConcurrentEndpointSyncs int32
-	// concurrentRSSyncs is the number of replica sets that are  allowed to sync
-	// concurrently. Larger number = more responsive replica  management, but more
-	// CPU (and network) load.
-	ConcurrentRSSyncs int32
-	// concurrentRCSyncs is the number of replication controllers that are
-	// allowed to sync concurrently. Larger number = more responsive replica
-	// management, but more CPU (and network) load.
-	ConcurrentRCSyncs int32
-	// concurrentServiceSyncs is the number of services that are
-	// allowed to sync concurrently. Larger number = more responsive service
-	// management, but more CPU (and network) load.
-	ConcurrentServiceSyncs int32
-	// concurrentResourceQuotaSyncs is the number of resource quotas that are
-	// allowed to sync concurrently. Larger number = more responsive quota
-	// management, but more CPU (and network) load.
-	ConcurrentResourceQuotaSyncs int32
-	// concurrentDeploymentSyncs is the number of deployment objects that are
-	// allowed to sync concurrently. Larger number = more responsive deployments,
-	// but more CPU (and network) load.
-	ConcurrentDeploymentSyncs int32
 	// concurrentDaemonSetSyncs is the number of daemonset objects that are
 	// allowed to sync concurrently. Larger number = more responsive daemonset,
 	// but more CPU (and network) load.
 	ConcurrentDaemonSetSyncs int32
```

Comment:
this belong to a DaemonSetControllerConfiguration, doesn't it?

Reply:
@stewart-yu can you double check all remaining fields here whether they belong to some controller?